### ISSUE TITLE
Explorer::makeRequest(): Wrong URL generation for APIs with absolute URLs

### ISSUE THREAD

**Issue by ureimers** [Jmeyering/hal-explorer#8]
I'm working with an API that exposes absolute URLs instead of relative ones, like:

`"doc:documents" : { "href" : "https://customer.com/rest/api/documents" }`

The base URL is `https://customer.com/rest/api`.

When I now call `$explorer->getRelation($response, 'documents')` it creates the URI `https://customer.com/rest/api//rest/api/documents` because of the way `Exploder::makeRequest()` assumes that the path given is relative.

I would suggest something like:

``` php
        // Remove the base url from the uri if it's an absolute one
        $parsed = parse_url(str_replace($this->getBaseUrl(), '', $uri));

        $path = $parsed["path"];

        // Trim leading slash if it exists
        $path = ltrim($path, "/");
```

Doing the `ltrim` later makes sure that only slashes from the extracted path are removed. Also this way network paths like `//github.com/rest/api` won't be broken.

**Comment by Jmeyering** [comment]
There is an interesting side effect here. The assumption that all absolute URL's are internal urls and it is safe to simple strip out the base url.

If my api is hosted at customer.com and my response contains

`"href": "http://othersite.com/resource"`

I want to follow that link absolutely.  I need to think of a good solution that works for this scenario also.

**Comment by Jmeyering** [comment]
Fixed by 1.4.0

**Comment by ureimers** [comment]
Oh yes, you're right! Ah, the devil is always in the detail.

Thanks for looking into it.
